doc/design/python/user_interface.md

```diff
@@ -0,0 +1,370 @@
+# User Interface Design
+
+## Basic Concepts
+### Variable
+A `Variable` represents shared, persistent state manipulated by a Paddle program.
+
+Variables are maintained by `pd.Variable` class,
+each `pd.Variable` represents a tensor whose value can be changed by ops.
+
+A basic way to create a variable is
+
+```python
+import paddle as pd
+
+v = pd.Variable(shape=[20, 20])
+```
+
+To make it more converient to share a variable, each `pd.Variable` has a name, 
+one can use a name to get or create a `pd.Variable` by calling `pd.get_variable`, for example:
+
+```python
+# same as 
+v = pd.get_variable(name="v", shape=[20, 20])
```

Comment:
get_varibale from what scope?

Reply:
get_variable from current namespace, in user's model config, they need only one global scope, but different namespace(add prefix to var's name).

The difference between multiple scope and namespace is that scopes will turn into a forest, different sub-scopes without the same ancestor is totally separated, but namespace doesn't suffer from this:

- all the variable are located in a global scope
- different namespaces have a different name prefix